### PR TITLE
add task solution

### PR DESCRIPTION
**Pull request by KolomietsN**
[DEMO LINK](https://KolomietsN.github.io/react_redux-list-of-posts/)

### CODE REVIEW

File: src/components/PostDetails.tsx [hunk 1]
```diff
@@ -104,23 +95,23 @@ export const PostDetails: React.FC<Props> = ({ post }) => {
...
 
-        {loaded && hasError && (
+        {!isLoading && hasError && (
           <div className="notification is-danger" data-cy="CommentsError">
             Something went wrong
           </div>
         )}
 
-        {loaded && !hasError && comments.length === 0 && (
+        {!isLoading && !hasError && comments.length === 0 && (
```
Comment (lerastarynets) [inline src/components/PostDetails.tsx:108]:
```suggestion
        {!isLoading && !hasError && !comments.length && (
```

File: src/components/PostDetails.tsx [hunk 2]
```diff
@@ -104,23 +95,23 @@ export const PostDetails: React.FC<Props> = ({ post }) => {
...
           <p className="title is-4" data-cy="NoCommentsMessage">
             No comments yet
           </p>
         )}
 
-        {loaded && !hasError && comments.length > 0 && (
+        {!isLoading && !hasError && comments.length > 0 && (
```
Comment (lerastarynets) [inline src/components/PostDetails.tsx:114]:
```suggestion
        {!isLoading && !hasError && !!comments.length && (
```

File: src/components/PostDetails.tsx [hunk 3]
```diff
@@ -154,7 +145,7 @@ export const PostDetails: React.FC<Props> = ({ post }) => {
           </>
         )}
 
-        {loaded && !hasError && !visible && (
+        {!isLoading && !hasError && !visible && (
```
Comment (lerastarynets) [inline src/components/PostDetails.tsx:148]: consider moving long ternaries to a separate const with consistent name and use the const here

**Review by lerastarynets — CHANGES_REQUESTED** [review]
Nice start! Let's fix and improve your work a little bit)

File: src/components/PostDetails.tsx [hunk 4]
```diff
@@ -2,33 +2,41 @@ import React, { useEffect, useState } from 'react';
...
+    isLoading,
+    hasError,
+  } = useAppSelector(state => state.comments);
+  const dispatch = useAppDispatch();
 
   function loadComments() {
-    setLoaded(false);
-    setError(false);
+    // setLoaded(false);
+    // setError(false);
```
Comment (anastasiiavorobiova) [inline src/components/PostDetails.tsx:28]:
```suggestion

```

**Review by anastasiiavorobiova — CHANGES_REQUESTED** [review]
Great work! Consider reading carefully the previous review comments and fix them 
Screenshot 2023-12-07 at 16 31 37
A user shouldn't be able to post comments with spaces only

**Review by VitaliyBondarenko1982 — CHANGES_REQUESTED** [review]
Please fix prev comment about creating comment only with spaces.
Don't see any commits after last review and still can to make comment just wiht spaces
Screenshot from 2023-12-07 17-48-30

**Pull request closed without merging**